### PR TITLE
fix: ensure pod security label on namespace

### PR DESCRIPTION
**Pull request by saumeya**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind bug
> /kind cleanup
> /kind failing-test
> /kind enhancement
> /kind documentation
> /kind code-refactoring

**What does this PR do / why we need it**:

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**Test acceptance criteria**:

* [ ] Unit Test
* [ ] E2E Test

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by svghadi** [comment]
/hold

We need to revisit this. Adding a restrictive pod security standard to an existing namespace (during an upgrade) will result in failure because the existing pods in the namespace won't be compliant with the restricted policy. The required policy configurations were added in https://github.com/argoproj-labs/argocd-operator/pull/1288 and https://github.com/argoproj-labs/argocd-operator/pull/1493, but these changes only work for fresh installs. Existing pods are not configured with the new policy settings unless the corresponding deployments are manually deleted.

**Review by svghadi — COMMENTED** [review]
OpenShift automatically handles the `securityContext` if it is missing. After running some tests, my previous concerns have been addressed, and there shouldn’t be any issues during the upgrade. I also have [PR #1533](https://github.com/argoproj-labs/argocd-operator/pull/1533) to handle existing deployments, but it’s not a blocker for this PR.

/remove-unhold

File: controllers/gitopsservice_controller.go [hunk 1]
```diff
@@ -920,3 +937,20 @@ func policyRuleForBackendServiceClusterRole() []rbacv1.PolicyRule {
...
+
+	namespace.Labels["pod-security.kubernetes.io/enforce"] = "restricted"
+	namespace.Labels["pod-security.kubernetes.io/enforce-version"] = "v1.29"
+	namespace.Labels["pod-security.kubernetes.io/audit"] = "restricted"
+	namespace.Labels["pod-security.kubernetes.io/audit-version"] = "latest"
+	namespace.Labels["pod-security.kubernetes.io/warn"] = "restricted"
+	namespace.Labels["pod-security.kubernetes.io/warn-version"] = "latest"
+
+	return true, namespace
+}
```
[the remark this answers is not in the document]
Comment (saumeya) [reply controllers/gitopsservice_controller.go:956]: Sure

**Review by svghadi — APPROVED** [review]
/lgtm

Thanks

File: controllers/gitopsservice_controller.go [hunk 2]
```diff
@@ -369,6 +377,15 @@ func (r *ReconcileGitopsService) reconcileDefaultArgoCDInstance(instance *pipeli
 				return reconcile.Result{}, err
 			}
 		}
+
+		needUpdate, updateNameSpace := ensurePodSecurityLabels(argocdNS)
```
Comment (iam-veeramalla) [inline controllers/gitopsservice_controller.go:381]: Just a NIT, I would change the word `needUpdate` to `needsUpdate`.

**Review by iam-veeramalla — COMMENTED** [review]
Thanks for the PR, LGTM except for the NIT.

**Pull request merged**